core/node/node_framework/src/implementations/layers/query_eth_client.rs

```diff
@@ -1,39 +1,74 @@
 use anyhow::Context;
-use zksync_types::{url::SensitiveUrl, SLChainId};
-use zksync_web3_decl::client::Client;
+use zksync_types::{url::SensitiveUrl, L2ChainId, SLChainId};
+use zksync_web3_decl::client::{Client, L1, L2};
 
 use crate::{
-    implementations::resources::eth_interface::EthInterfaceResource,
+    implementations::resources::eth_interface::{EthInterfaceResource, L2InterfaceResource},
+    service::ServiceContext,
     wiring_layer::{WiringError, WiringLayer},
+    IntoContext,
 };
 
 /// Wiring layer for Ethereum client.
 #[derive(Debug)]
 pub struct QueryEthClientLayer {
     chain_id: SLChainId,
     web3_url: SensitiveUrl,
+    l2_mode: bool,
 }
 
 impl QueryEthClientLayer {
-    pub fn new(chain_id: SLChainId, web3_url: SensitiveUrl) -> Self {
-        Self { chain_id, web3_url }
+    pub fn new(chain_id: SLChainId, web3_url: SensitiveUrl, l2_mode: bool) -> Self {
+        Self {
+            chain_id,
+            web3_url,
+            l2_mode,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct Output {
+    query_client_l1: Client<L1>,
+    query_client_l2: Option<Client<L2>>,
+}
+
+impl IntoContext for Output {
```

Comment:
Why not declare output as containing resources directly?

```rust
#[derive(Debug, IntoContext)]
#[context(crate = crate)]
pub struct Output {
    query_client_l1: EthInterfaceResource,
    query_client_l2: Option<L2InterfaceResource>,
}
```

This is the approach used by other wiring layers.

Reply:
Makes sense, I think I did not notice that the `Option<X>` is possible there